### PR TITLE
Deprecate All

### PR DESCRIPTION
**Pull request by bkamins**

### CODE REVIEW

**Comment by bkamins** [comment]
Fixes https://github.com/JuliaData/DataAPI.jl/issues/27

**Comment by bkamins** [comment]
I left the tests for `All` for the time being as is (I propose to remove them when we remove `All`).

File: src/DataAPI.jl [hunk 1]
```diff
@@ -121,7 +121,14 @@ Select the union of the selections in `cols`. If `cols == ()`, select all column
 """
 struct All{T<:Tuple}
     cols::T
-    All(args...) = new{typeof(args)}(args)
+    function All(args...)
+        if isempty(args)
+            Base.depwarn("All() is deprecated, use Cols(:) instead", :All)
+        else
```
Comment (bkamins) [inline src/DataAPI.jl:127]:
```suggestion
        if !isempty(args)
```

**Comment by bkamins** [comment]
per recommendation of @piever I updated the PR to allow `All()` as I think it is not a problem to keep it.

**Review by nalimilan — APPROVED** [review]
Looks good, though better wait a bit to see whether IndexedTable maintainers are OK with deprecating `All()` too.

**Comment by bkamins** [comment]
It seems we will not get much feedback here. Given `All()` is a natural name I would recommend to retain `All()` but deprecate `All(args...)`.

**Review by piever — APPROVED** [review]

**Comment by bkamins** [comment]
@nalimilan - do we go ahead and merge this then?

**Comment by bkamins** [comment]
Thank you. Will you tag a release?

**Pull request merged**